### PR TITLE
sshd service: Default to INFO logLevel (upstream default)

### PR DESCRIPTION
**Pull request by nh2**
###### Motivation for this change

This brings NixOS in line with upstream and other Linux distributions, and reduces log spam on servers due to bruteforcing botnets.

The previous justification (added in #40692) for using `VERBOSE` is incorrect, because OpenSSH does use level `INFO` to log "which key was used to log in" for sccessful logins, see:

https://github.com/openssh/openssh-portable/blob/6247812c76f70b2245f3c23f5074665b3d436cae/auth.c#L323-L328

I've verified it; sshd logs with `VERBOSE` look like:

```
sshd: Connection from 192.0.2.1 port 45668 on 192.0.2.2 port 22 rdomain ""
sshd: Accepted key RSA SHA256:Cabc123abc123abc12/abc123abc123abc123abc123 found at /etc/ssh/authorized_keys.d/root:1
sshd: Postponed publickey for root from 192.0.2.1 port 45668 ssh2 [preauth]
sshd: Accepted key RSA SHA256:Cabc123abc123abc12/abc123abc123abc123abc123 found at /etc/ssh/authorized_keys.d/root:1
sshd: Accepted publickey for root from 192.0.2.1 port 45668 ssh2: RSA SHA256:Cabc123abc123abc12/abc123abc123abc123abc123
sshd: pam_unix(sshd:session): session opened for user root by (uid=0)
systemd-logind: [🡕] New session 12 of user root.
```

and with `INFO` look like:

```
sshd: Accepted publickey for root from 192.0.2.1 port 45694 ssh2: RSA SHA256:Cabc123abc123abc12/abc123abc123abc123abc123
sshd: pam_unix(sshd:session): session opened for user root by (uid=0)
systemd-logind: [🡕] New session 12 of user root.
```

As shown, the fingerprint for the successful login is still printed with `INFO`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

CC @bjornfor @volth @Izorkin from #40692.

CC recent `sshd` committers @aanderse @Infinisil @jeaye 

CC @eelco @lovek323 @fpletz as `fail2ban` maintainers, and @c0bw3b @Baughn as recent `fail2ban` service committers.

### CODE REVIEW

**Comment by nh2** [comment]
I got the bit about `fail2ban` from https://wiki.archlinux.org/index.php/fail2ban#Custom_SSH_jail which says:

> It may be necessary to set `LogLevel VERBOSE` in `/etc/ssh/sshd_config` to allow full fail2ban monitoring as otherwise password failures may not be logged correctly.

**Review by RaitoBezarius — APPROVED** [review]

**Comment by Izorkin** [comment]
Also try adding to `fail2ban` service this variant  - `services.openssh.logLevel = mkDefault "VERBOSE";`

File: nixos/modules/services/networking/ssh/sshd.nix
Comment (aneeshusa) [inline nixos/modules/services/networking/ssh/sshd.nix:333]:
```suggestion
          QUIET, FATAL, ERROR, INFO, VERBOSE, DEBUG, DEBUG1, DEBUG2, and DEBUG3. The default is INFO. DEBUG and DEBUG1
```

**Review by aneeshusa — APPROVED** [review]
SGTM as openssh maintainer though I didn't test

File: nixos/modules/services/security/fail2ban.nix
Comment (aanderse) [inline nixos/modules/services/security/fail2ban.nix:215]: The wording makes it sound like action is required here under any scenario.

Comment (nh2) [reply nixos/modules/services/security/fail2ban.nix]: I have improved the wording, switching it to a passive to make it clearer.

**Comment by nh2** [comment]
> Also try adding to `fail2ban` service this variant - `services.openssh.logLevel = mkDefault "VERBOSE";`

@Izorkin This looks like a nice idea, I implemented that.

I'll wait another half-a-month in case more feedback arrives, after that I'll probably merge it.

File: nixos/modules/services/security/fail2ban.nix
Comment (aanderse) [reply nixos/modules/services/security/fail2ban.nix]: The documentation still says I have to do something when I don't. The `fail2ban` takes care of it for me, but documentation says I have to.

**Review by Izorkin — APPROVED** [review]

File: nixos/modules/services/security/fail2ban.nix
Comment (nh2) [reply nixos/modules/services/security/fail2ban.nix]: @aanderse Right in the next sentence it says

```
          This module sets it to <literal>"VERBOSE"</literal> if
          not set otherwise, so enabling fail2ban can make SSH logs
          more verbose. 
```

indicating that you not have to do something unless you have otherwise set `services.openssh.logLevel`.

Comment (aanderse) [reply nixos/modules/services/security/fail2ban.nix]: Need more sleep :sleeping: Sorry for the hassle :wink:

**Review by aanderse — APPROVED** [review]

File: nixos/modules/services/security/fail2ban.nix
Comment (nh2) [reply nixos/modules/services/security/fail2ban.nix]: No prob :grin:

**Comment by nh2** [comment]
I fixed the merge conflicts and squashed the commits, as they better fit into one.

**Comment by aanderse** [comment]
Thanks @nh2!

**Pull request merged**
